addons/sale_project/models/sale_order.py

```diff
@@ -74,26 +76,30 @@ def _search_tasks_ids(self, operator, value):
     def _compute_tasks_ids(self):
         tasks_per_so = self.env['project.task']._read_group(
             domain=['&', ('project_id', '!=', False), '|', ('sale_line_id', 'in', self.order_line.ids), ('sale_order_id', 'in', self.ids)],
-            groupby=['sale_order_id'],
+            groupby=['sale_order_id', 'state'],
             aggregates=['id:recordset', '__count']
         )
         so_with_tasks = self.env['sale.order']
-        for order, tasks_ids, tasks_count in tasks_per_so:
+        for order, state, tasks_ids, tasks_count in tasks_per_so:
             if order:
-                order.tasks_ids = tasks_ids
-                order.tasks_count = tasks_count
+                order.tasks_ids += tasks_ids
+                order.tasks_count += tasks_count
+                order.closed_task_count += state in ['1_done', '1_canceled'] and tasks_count
                 so_with_tasks += order
             else:
                 # tasks that have no sale_order_id need to be associated with the SO from their sale_line_id
                 for task in tasks_ids:
                     task_so = task.sale_line_id.order_id
                     task_so.tasks_ids = [Command.link(task.id)]
                     task_so.tasks_count += 1
+                    if (task.state in ['1_done', '1_canceled']):
+                        task_so.closed_task_count += 1
```

Comment:
```suggestion
                    task_so.closed_task_count += state in ['1_done', '1_canceled']
```
We can apply the same format as L87.
Also `task.state == state`, no?